### PR TITLE
fillers: convert the eip3860 test fillers to pytest format

### PR DESCRIPTION
**Pull request by danceratopz**

### CODE REVIEW

File: fillers/eips/eip3860.py [hunk 1]
```diff
@@ -8,11 +8,13 @@
...
+
+from ethereum_test_forks import Fork, Shanghai, forks_from
 from ethereum_test_tools import (
     Account,
     Block,
-    BlockchainTest,
+    BlockchainTestFiller,
     Environment,
     Initcode,
     StateTest,
```
Comment (marioevz) [inline fillers/eips/eip3860.py:20]:
```suggestion
    StateTestFiller,
```

File: fillers/eips/eip3860.py [hunk 2]
```diff
@@ -233,504 +259,394 @@ def generate_tx_initcode_limit_test_cases(
...
-            nonce=1,
-        ),
-        to_address(0x200): Account(
-            code=call_code,
-            nonce=1,
-        ),
-    }
+    def test_create_opcode_initcode(
+        self,
+        state_test: StateTest,
```
Comment (marioevz) [inline fillers/eips/eip3860.py:538]:
```suggestion
        state_test: StateTestFiller,
```

File: fillers/eips/eip3860.py [hunk 3]
```diff
@@ -233,504 +259,394 @@ def generate_tx_initcode_limit_test_cases(
...
-    else:
-        tx = Transaction(
+    @pytest.fixture
+    def pre(self) -> Dict[Any, Any]:  # noqa: D102
+        return {
+            TestAddress: Account(balance=1000000000000000000000),
+        }
+
+    @pytest.fixture
+    def tx_error(self, gas_test_case) -> str:
```
Comment (marioevz) [inline fillers/eips/eip3860.py:346]:
```suggestion
    def tx_error(self, gas_test_case) -> str | None:
```

File: fillers/eips/eip3860.py [hunk 4]
```diff
@@ -233,504 +259,394 @@ def generate_tx_initcode_limit_test_cases(
...
+                        "create2(0, 0, contract_length, 0xDEADBEEF)"
+                    )
+                )
+            )
+        raise Exception("invalid opcode for generator")
 
-    else:
+    @pytest.fixture
+    def created_contract_address(  # noqa: D102
+        self, initcode: Initcode, opcode: str
```
Comment (marioevz) [inline fillers/eips/eip3860.py:521]:
```suggestion
        self, initcode: Initcode, opcode: Op
```

File: fillers/eips/eip3860.py [hunk 5]
```diff
@@ -233,504 +259,394 @@ def generate_tx_initcode_limit_test_cases(
...
+        EMPTY_INITCODE,
+        SINGLE_BYTE_INITCODE,
+        INITCODE_ZEROS_32_BYTES,
+        INITCODE_ZEROS_33_BYTES,
+        INITCODE_ZEROS_49120_BYTES,
+        INITCODE_ZEROS_49121_BYTES,
+    ],
+    ids=get_initcode_name,
+)
+@pytest.mark.parametrize("opcode", ["create", "create2"], ids=lambda x: x)
```
Comment (marioevz) [inline fillers/eips/eip3860.py:481]: I think we should parametrize here directly using the `Op.CREATE` and `Op.CREATE2`.

File: fillers/eips/eip3860.py [hunk 6]
```diff
@@ -233,504 +259,394 @@ def generate_tx_initcode_limit_test_cases(
...
+            {
+                let contract_length := calldatasize()
+                calldatacopy(0, 0, contract_length)
+                let gas1 := gas()
+                let res := ${create_instruction}
+                let gas2 := gas()
+                sstore(0, res)
+                sstore(1, sub(gas1, gas2))
+            }
+            """
```
Comment (marioevz) [inline fillers/eips/eip3860.py:501]:
```suggestion
        create_code = (
            Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE)
            + Op.GAS
            + create_call
            + Op.GAS
            # stack: [Gas 2, Call Result, Gas 1]
            + Op.SWAP1
            # stack: [Call Result, Gas 2, Gas 1]
            + Op.SSTORE(0)
            # stack: [Gas 2, Gas 1]
            + Op.SWAP1
            # stack: [Gas 1, Gas 2]
            + Op.SUB
            # stack: [Gas 1 - Gas 2]
            + Op.SSTORE(1)
        )
```
We can rewrite as such, where `create_call` depends on the `opcode` parameter. The only other change required is on line 580 with the gas calculation would be a bit affected by this.

**Review by marioevz — COMMENTED** [review]
Looks great! Just a couple of suggestions. Thanks!

File: fillers/eips/eip3860.py (hunk 5 above)
Comment (danceratopz) [reply fillers/eips/eip3860.py:481]: Definitely more explicit and cleaner.

File: fillers/eips/eip3860.py (hunk 6 above)
Comment (danceratopz) [reply fillers/eips/eip3860.py]: Refactored in ec3490c07f25ad00845d6b5a27ed7f886bc5631e. Could you please check the `create_call` and gas calculation logic?

**Review by marioevz — APPROVED** [review]
Looks great!
I'm not sure why github is detecting 29 files changes though. If I compare locally there are only three files changed, so there might be something off with the branching?

**Comment by danceratopz** [comment]
Superseded by #13.

**Pull request closed without merging**